ext/reflection/php_reflection.c

```diff
@@ -3089,18 +3089,18 @@ ZEND_METHOD(ReflectionMethod, __toString)
 ZEND_METHOD(ReflectionMethod, getClosure)
 {
 	reflection_object *intern;
-	zval *obj;
+	zval *obj = NULL;
 	zend_function *mptr;
 
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|o!", &obj) == FAILURE) {
+		RETURN_THROWS();
+	}
+
 	GET_REFLECTION_OBJECT_PTR(mptr);
 
 	if (mptr->common.fn_flags & ZEND_ACC_STATIC)  {
 		zend_create_fake_closure(return_value, mptr, mptr->common.scope, mptr->common.scope, NULL);
-	} else {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "o", &obj) == FAILURE) {
-			RETURN_THROWS();
-		}
-
+	} else if (obj) {
```

Comment:
This doesn't look right to me. The intention here is that the object is required if the method is non-static. If this kind of change is made, then this needs to error if obj is null.